lib/spec/openapi/utils.js

```diff
@@ -167,6 +167,9 @@ function plainJsonObjectToOpenapi3 (container, jsonSchema, externalSchemas) {
             type: jsonSchemaElement.type
           }
         }
+
+        if (jsonSchemaElement.$ref) result.schema.$ref = jsonSchemaElement.$ref
```

Comment:
I'm wondering whether this should be
```ts
if (jsonSchemaElement.$ref) {
   result.schema.$ref = jsonSchemaElement.$ref
   delete result.schema.type
   delete result.description
 }
```

To avoid `undefined` fields in `fastify.swagger()`?

Reply:
As you pass a `$ref` the property `type` has no purpose to exist, on the other hand `description` is still legit to be here. So technically you should only `delete result.schema.type`